### PR TITLE
[enhencement](jdbc catalog) Use Druid instead of HikariCP in JdbcClient

### PR DESCRIPTION
**Pull request by BePPPower**
# Proposed changes

Issue Number: close #xxx

This pr does three things:
1. Use Druid instead of HikariCP in JdbcClient
2. when download udf jar, add the name of the jar package after the local file name.
3. refactor some jdbcResource code

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by CalvinKirs** [comment]
>> Use Druid instead of HikariCP in JdbcClient
Why are you doing this? It's no make sense to me.

File: be/src/vec/exec/vjdbc_connector.cpp [hunk 1]
```diff
@@ -140,6 +140,7 @@ Status JdbcConnector::open(RuntimeState* state, bool read) {
                     std::abs((int64_t)hash_str(_conn_param.resource_name)), _conn_param.driver_path,
                     _conn_param.driver_checksum, &local_location));
         }
+        LOG(INFO) << "driver local path = " << local_location;
```
Comment (morningman) [inline be/src/vec/exec/vjdbc_connector.cpp:143]: Use VLOG

**Comment by morningman** [comment]
> 

Unify to use same logic as we did in `JdbcExecutor`

File: be/src/vec/exec/vjdbc_connector.cpp (hunk 1 above)
Comment (BePPPower) [reply be/src/vec/exec/vjdbc_connector.cpp]: done

**Comment by BePPPower** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.32 seconds
 stream load tsv:          452 seconds loaded 74807831229 Bytes, about 157 MB/s
 stream load json:         36 seconds loaded 2358488459 Bytes, about 62 MB/s
 stream load orc:          74 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230306120806_clickbench_pr_109350.html

**Review by morningman — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Comment by zjf88865** [comment]
I don't recommend using Druid to replace HikariCP. HikariCP has much higher performance than Druid, and I think this requirement is invalid. I don't recommend adopting it. If monitoring or similar tasks are needed, they can be done through third-party middleware. Don't affect Doris' performance. I really don't understand why such a strange requirement was proposed. Even Spring uses HikariCP by default. If Druid is used, it's like putting the cart before the horse. I think this is also why Doris does not use Druid by default, so I don't recommend adopting it

**Pull request merged**